modules/le_log/le_log.h

```diff
@@ -0,0 +1,103 @@
+#ifndef GUARD_le_log_H
+#define GUARD_le_log_H
+
+#include "le_core/le_core.h"
+
+namespace le_log {
+enum class Level : uint8_t {
+	DEBUG = 0,
+	INFO  = 1,
+	WARN  = 2,
+	ERROR = 3
+};
+}
+
+struct le_log_module_o;
+struct le_log_context_o;
+
+// clang-format off
+struct le_log_api {
+
+    void ( *set_level  )(le_log_module_o *module, le_log::Level level);
```

Comment:
i feel it would be helpful to have a logger interface here, as we do with other modules.

there might be some le_log related functionality which is not related to individual `le_log_module_o` objects, (as in, for example a possibly future feature to silence any log messages below a certain log level), and it will then be easier to keep things organised if we keep all le_log_module_o* related functions in their own interface. 

We can think of interfaces as the class namespace for functions which can be applied on `le_log_module_o` objects. placing this inside of a struct helps us in another way: it provides a namespace to clients of the c api. 

My hope here is that it will help with consistency and discoverability, as methods on objects from `le_module`s are usually called via their object's interface, and not their parent api interface.

I suggest naming the interface `le_log_module_interface_t` and grouping L21-31 under this interface...

Reply:
can do